tests/callbacks/test_callbacks.py

```diff
@@ -13,285 +13,108 @@
-    assert test_callback.on_init_start_called
-    assert test_callback.on_init_end_called
-    assert test_callback.on_fit_start_called
-    assert test_callback.on_fit_end_called
-    assert test_callback.on_sanity_check_start_called
-    assert test_callback.on_sanity_check_end_called
-    assert test_callback.on_epoch_start_called
-    assert test_callback.on_epoch_start_called
-    assert test_callback.on_batch_start_called
-    assert test_callback.on_batch_end_called
-    assert test_callback.on_train_batch_start_called
-    assert test_callback.on_train_batch_end_called
-    assert test_callback.on_validation_batch_start_called
-    assert test_callback.on_validation_batch_end_called
-    assert test_callback.on_train_start_called
-    assert test_callback.on_train_end_called
-    assert test_callback.on_pretrain_routine_start_called
-    assert test_callback.on_pretrain_routine_end_called
-    assert test_callback.on_validation_start_called
-    assert test_callback.on_validation_end_called
-    assert not test_callback.on_test_batch_start_called
-    assert not test_callback.on_test_batch_end_called
-    assert not test_callback.on_test_start_called
-    assert not test_callback.on_test_end_called
-    assert test_callback.on_after_backward_called
-    assert test_callback.on_before_zero_grad_called
-
-    # reset setup teardown callback
-    test_callback.teardown_called = False
-    test_callback.setup_called = False
-
-    test_callback = TestCallback()
-    trainer_options.update(callbacks=[test_callback])
-    trainer = Trainer(**trainer_options)
+    # check if a method was called exactly once
+    callback.on_fit_start.assert_called_once()
+
+    # check how many times a method was called
+    assert callback.on_train_batch_end.call_count == 1
+
+    # check that a method was NEVER called
+    callback.on_keyboard_interrupt.assert_not_called()
+
+    # check with what a method was called
+    callback.on_fit_end.assert_called_with(trainer, model)
+
+    # check exact call order
+    callback.assert_has_calls([
+        call.on_init_start(trainer),
+        call.on_init_end(trainer),
+        call.setup(trainer, None, "fit"),
+        call.on_fit_start(trainer, model),
+        call.on_pretrain_routine_start(trainer, model),
+        call.on_pretrain_routine_end(trainer, model),
+
+        # sanity check start
+        call.on_sanity_check_start(trainer, model),
+        call.on_validation_start(trainer, model),
+        call.on_validation_epoch_start(trainer, model),
+        call.on_validation_batch_start(trainer, model, ANY, 0, 0),
+        call.on_validation_batch_end(trainer, model, ANY, ANY, 0, 0),
+        call.on_validation_epoch_end(trainer, model),
+        call.on_validation_end(trainer, model),
+        call.on_sanity_check_end(trainer, model),
+        # # sanity check end
+
+        # train start
+        call.on_train_start(trainer, model),
+        call.on_epoch_start(trainer, model),
```

Comment:
should we make these changes in a separate one? first we need to decide whether we should update this https://github.com/PyTorchLightning/pytorch-lightning/issues/4310#issuecomment-715306697 accordingly or not?

Reply:
nice. could you ping me on this for review (in separate one or here, whatever you decide) 
thanks!

Reply:
I decided to change the test with mock here, no update
And we can update the hooks after that and they can easily swapped

Reply:
Actually I had something else in mind with this test. Someting like checking when they are called using a rank or something. But yeah we can change this test again later sometime.